File: BookWorm/BookWorm.Services/DataBase/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookWorm.Services.DataBase
{
    public class Book
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(255)]
        public string Title { get; set; } = string.Empty;
        
        public int AuthorId { get; set; }

        [ForeignKey("AuthorId")]
        public Author Author { get; set; } = null!;
        
        [MaxLength(1000)]
        public string Description { get; set; }=string.Empty;

        public int PublicationYear { get; set; }
        
        public int PageCount { get; set; }
        
   
        public string? CoverImagePath { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
        
        public int? CreatedByUserId { get; set; }
        
        [ForeignKey("CreatedByUserId")]
        public User? CreatedByUser { get; set; }
       
        public virtual ICollection<BookGenre> BookGenres { get; set; } = new List<BookGenre>();
        public virtual ICollection<BookReview> BookReviews { get; set; } = new List<BookReview>();
        public virtual ICollection<ReadingListBook> ReadingListBooks { get; set; } = new List<ReadingListBook>();
        public virtual ICollection<ReadingChallengeBook> ReadingChallengeBooks { get; set; } = new List<ReadingChallengeBook>();
        [MaxLength(1000)]
        public string BookState { get; set; } = "Initial";
    }
} 

File: BookWorm/UI/bookworm_mobile/lib/model/book.dart
import 'package:json_annotation/json_annotation.dart';

part 'book.g.dart';
@JsonSerializable()
class Book {
  final int id;
  final String title;
  final int authorId;
  final String authorName;
  final String description;
   final int publicationYear;
    final int pageCount;
  final String bookState;
  final List<String> genres;
  final String? coverImagePath;

  Book({
    this.id = 0,
    this.title = '',
    this.authorId = 0,
    this.authorName = '',
    this.bookState = 'Accepted',
    this.publicationYear = 0,
    this.pageCount = 0,
    this.description = '',
    this.genres = const [],
    this.coverImagePath,
  });
  factory Book.fromJson(Map<String, dynamic> json) => _$BookFromJson(json);
}